File: 5/perl/5.pl
#!/usr/bin/perl

use strict;
use warnings;

my $x = 2520;
my $start;
BEGIN
{
  $start = time;
}
END
{
  print "Time: " . (time - $start) . "\n";
}
while(1)
{
  my $t = 1;
  for (1..20)
  {
    if($x % $_)
    {
      $t = 0;
      last;
    }
  }
  if ($t)
  {
    print "Answer: $x\n";
    exit;
  }
  $x += 2;
}

File: 5/cpp/5.cpp
#include<cstdio>
#include<ctime>

int main()
{
  std::time_t start = std::time(0);

  long x = 2520;

  while(1)
  {
    int t = 1;
    for(int i = 1; i <= 20; i++)
    {
      if(x % i)
      {
        t = 0;
        break;
      }
    }
    if(t)
    {
      printf("Answer: %d\n", x);
      std::time_t end = std::time(0);
      printf("Time: %d\n", end - start);
      return(0);
    }
    x += 2;
  } 
}
